model-sync-lib/src/test/kotlin/org/modelix/model/sync/ModelImporterTest.kt

```diff
@@ -0,0 +1,96 @@
+package org.modelix.model.sync
+
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertDoesNotThrow
+import org.modelix.model.api.IBranch
+import org.modelix.model.api.PBranch
+import org.modelix.model.api.getRootNode
+import org.modelix.model.client.IdGenerator
+import org.modelix.model.data.ModelData
+import org.modelix.model.data.NodeData
+import org.modelix.model.data.asData
+import org.modelix.model.lazy.CLTree
+import org.modelix.model.lazy.ObjectStoreCache
+import org.modelix.model.persistent.MapBaseStore
+import java.io.File
+
+class ModelImporterTest {
+
+    companion object {
+        private lateinit var model: ModelData
+        private lateinit var newModel: ModelData
+        private lateinit var branch: IBranch
+
+        @JvmStatic
+        @BeforeAll
+        fun `load and import model`() {
+            model = ModelData.fromJson(File("src/test/resources/model.json").readText())
+            val newModelFile = File("src/test/resources/newmodel.json")
```

Comment:
In addition to this hand written test data I would also use same randomly generated input to cover more corner cases. See `RandomTreeChangeGenerator` and its usages for an example. It's good at finding all the cases that you would never think about.